substrate/client/api/src/lib.rs

```diff
@@ -69,10 +69,8 @@ pub mod utils {
 				return Ok(false)
 			}
 
-			let current = current.as_ref().map(|(c, p)| (c.borrow(), p.borrow()));
-
 			let mut hash = hash;
-			if let Some((current_hash, current_parent_hash)) = current {
+			if let Some((current_hash, current_parent_hash)) = current.as_ref() {
```

Comment:
This should just work too:

```suggestion
			if let Some((current_hash, current_parent_hash)) = &current {
```